desktop/test/e2e/specs/login.js

```diff
@@ -11,6 +11,15 @@ switch ( process.platform ) {
 		APP_PATH = path.join( __dirname, '../../../release/linux-unpacked/wpcom' );
 		break;
 	case 'darwin':
+		// On Apple Silicon, the output file is under a separate directory.
+		if ( process.arch.includes( 'arm' ) ) {
+			APP_PATH = path.join(
+				__dirname,
+				'../../../release/mac-arm64/WordPress.com.app/Contents/MacOS/WordPress.com'
+			);
+			break;
+		}
+		// Codepath for Intel architecture.
```

Comment:
Nice. 👍🏽 Having this in the code is useful. 

However as I noted in my other comment, it's unclear which architecture CI is using for the e2e run. 🤔 My preference would be to build only one architecture for non-release CI runs, but if building both you would want to make it so the test suite is actually ran against both.

Reply:
CI will use whatever architecture it detects in `process.arch`.

On Intel builders, it would be the x64 binaries. On Apple Silicon builders, it would be the arm64 binaries.

The issue this addresses, is that previously were were using Intel builders to build x64 and arm64 builds. It worked fine when the primary build we are producing was x64, but as soon as we defaulted to producing arm64 the E2E tests can no longer execute because arm64 is architecturally not backward compatible with x64.

Reply:
One other thing to note is that having this conditional makes it possible for both Intel and Apple Silicon processors to build and run the E2E builds locally.

Reply:
> CI will use whatever architecture it detects in process.arch.

ah, yes - I missed this detail. 👍🏽 

> One other thing to note is that having this conditional makes it possible for both Intel and Apple Silicon processors to build and run the E2E builds locally.

Yep - this is a nice change to the script.